target/linux/ramips/dts/mt7628an_hak5_wifi-pineapple-mk7.dts

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an.dtsi"
+
+/ {
+	compatible = "hak5,wifi-pineapple-mk7", "mediatek,mt7628an-soc";
+	model = "Hak5 WiFi Pineapple Mark 7";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x08000000>;
+	};
+
+	aliases {
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		red {
+			label = "red:system";
+			gpios = <&gpio 0 0>;
+		};
+
+		green {
+			label = "green:system";
+			gpios = <&gpio 2 0>;
+		};
+
+		led_status: blue {
+			label = "blue:system";
+			gpios = <&gpio 3 0>;
+		};
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "p4led_an", "p0led_an", "p1led_an", "p2led_an", "p3led_an", "wdt", "refclk", "i2s";
+			ralink,function = "gpio";
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xf80000>;
+			};
+		};
+	};
+};
+
+&ethernet {
+	status = "okay";
+	mtd-mac-address = <&factory 0x4>;
```

Comment:
Nothing in 0x28, 0x2e, 0xe000 or 0xe006?

Reply:
I'm relying on data given to me by the hardware manufacturer here, it seems to be working fine.

Reply:
I'm sure it's working. So, you know that vendor firmware does it exactly the same, i.e. uses the address in 0x4 for ethernet? Or is it just the same, but stored elsewhere?

Reply:
The vendor firmware does exactly the same, I authored the DTS being used in the vendor firmware and am now making the effort to mainline the devices into OpenWRT.

Reply:
Okay, then it's fine. Many vendors put ethernet addresses separately in the named locations as defined by the SDKs ...
This is then frequently overlooked when wifi and ethernet happen to have the same address, that's why I asked so persistently.